generator.py

```diff
@@ -0,0 +1,233 @@
+import argparse
+import subprocess
+import sys
+import asyncio
+
+DEFAULT_GIR_FILES_DIRECTORY = Path("./gir-files")
+DEFAULT_GIR_DIRECTORY = Path("./gir/")
+DEFAULT_GIR_PATH = DEFAULT_GIR_DIRECTORY / "target/release/gir"
+
+
+def run_command(command, folder=None):
+    return subprocess.run(command, cwd=folder, check=True)
+
+
+async def spawn_process(exe, args):
+    p = await asyncio.create_subprocess_exec(
+        exe,
+        *args,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+
+    stdout, stderr = await p.communicate()
+    assert p.returncode == 0, stderr.strip()
+    stdout = stdout.decode("utf-8")
+    stderr = stderr.decode("utf-8")
+    return stdout, stderr
+
+
+async def spawn_gir(gir_exe, args):
+    stdout, stderr = await spawn_process(gir_exe, args)
+    # Gir doesn't print anything to stdout. If it does, this is likely out of
+    # order with stderr, unless the printer/logging flushes in between.
+    assert not stdout, "`gir` printed unexpected stdout: {}".format(stdout)
+    return stderr
+
+
+def update_workspace():
+    return run_command(["cargo", "build", "--release"], "gir")
+
+
+def ask_yes_no_question(question, conf):
+    question = "{} [y/N] ".format(question)
+    if conf.yes:
+        print(question + "y")
+        return True
+    line = input(question)
+    return line.strip().lower() == "y"
+
+
+def update_submodule(submodule_path, conf):
+    if any(submodule_path.iterdir()):
+        return False
+    print("=> Initializing {} submodule...".format(submodule_path))
+    run_command(["git", "submodule", "update", "--init", submodule_path])
+    print("<= Done!")
+
+    if ask_yes_no_question(
+        "Do you want to update {} submodule?".format(submodule_path), conf
+    ):
+        print("=> Updating submodule...")
+        run_command(["git", "reset", "--hard", "HEAD"], submodule_path)
+        run_command(["git", "pull", "-f", "origin", "master"], submodule_path)
+        print("<= Done!")
+        return True
+    return False
+
+
+def build_gir():
+    print("=> Building gir...")
+    update_workspace()
+    print("<= Done!")
+
+
+async def regenerate_crate_docs(gir_exe, crate_dir, base_gir_args):
+    doc_path = "docs.md"
+    # Generate into docs.md instead of the default vendor.md
+    doc_args = base_gir_args + ["-m", "doc", "--doc-target-path", doc_path]
+
+    # The above `gir -m doc` generates docs.md relative to the directory containing Gir.toml
+    doc_path = crate_dir / doc_path
+    embed_args = ["-g", "-o", doc_path, "-d", crate_dir]
+
+    logs = "Regenerating documentation for `{}` into `{}`...\n".format(
+        crate_dir, doc_path
+    )
+    logs += await spawn_gir(gir_exe, doc_args)
+
+    logs += "Embedding documentation from `{}` into `{}`...\n".format(
+        doc_path, crate_dir
+    )
+    stdout, stderr = await spawn_process("rustdoc-stripper", embed_args)
+    logs += stdout
+    logs += stderr
+
+    return logs
```

Comment:
Perhaps we should delete `docs.md` again after embedding instead of leaving unused cruft?

Otherwise:
- Generate docs to `/tmp`?
- Allow `gir` to print to `stdout`, and `rustdoc-stripper` to read from `stdin`?

The problem here, again, is that `--doc-target-path` is relative to the directory containing `Gir.toml`.